ui-v2/src/api/flow-runs/index.ts

```diff
@@ -217,3 +232,83 @@ export const useDeploymentCreateFlowRun = () => {
 		...rest,
 	};
 };
+
+/**
+ * Hook for changing a flow run's state
+ *
+ * @returns Mutation object for setting a flow run state with loading/error states and trigger function
+ *
+ * @example
+ * ```ts
+ * const { setFlowRunState, isLoading } = useSetFlowRunState();
+ *
+ * setFlowRunState({
+ *   id: "flow-run-id",
+ *   state: { type: "COMPLETED" },
+ *   message: "State changed by user"
+ * });
+ * ```
+ */
+export const useSetFlowRunState = () => {
+	const queryClient = useQueryClient();
+	const { mutate: setFlowRunState, ...rest } = useMutation({
+		mutationFn: async ({ id, ...params }: SetFlowRunStateParams) => {
+			const res = await getQueryService().POST("/flow_runs/{id}/set_state", {
+				params: { path: { id } },
+				body: params,
+			});
+
+			if (!res.data) {
+				throw new Error("'data' expected");
+			}
+			return res.data;
+		},
+		onMutate: async ({ id, state }) => {
+			await queryClient.cancelQueries({ queryKey: queryKeyFactory.detail(id) });
+
+			const previousFlowRun = queryClient.getQueryData<FlowRun>(
+				queryKeyFactory.detail(id),
+			);
+
+			if (previousFlowRun?.state) {
+				queryClient.setQueryData<FlowRun>(queryKeyFactory.detail(id), {
+					...previousFlowRun,
+					state: {
+						id: previousFlowRun.state.id,
+						type: state.type,
+						name: state.name ?? previousFlowRun.state.name,
+						message: state.message ?? previousFlowRun.state.message,
+						timestamp: new Date().toISOString(),
+						data: previousFlowRun.state.data,
+						state_details: previousFlowRun.state.state_details,
+					},
+				});
+			}
+
+			return { previousFlowRun };
+		},
+		onError: (err, { id }, context) => {
+			// Roll back optimistic update on error
+			if (context?.previousFlowRun) {
+				queryClient.setQueryData(
+					queryKeyFactory.detail(id),
+					context.previousFlowRun,
+				);
+			}
+
+			throw err instanceof Error
+				? err
+				: new Error("Failed to update flow run state");
+		},
+		onSettled: () => {
+			void Promise.all([
+				queryClient.invalidateQueries({ queryKey: queryKeyFactory.all() }),
+				queryClient.invalidateQueries({ queryKey: queryKeyFactory.details() }),
+			]);
+		},
```

Comment:
I think you want something like this:
```suggestion
				onSettled: (_data, _error, { id }) => {
			void Promise.all([
				queryClient.invalidateQueries({ queryKey: queryKeyFactory.lists() }),
				queryClient.invalidateQueries({ queryKey: queryKeyFactory.detail(id) }),
			]);
		},

```
This will invalidate all lists and the detail query for just the corresponding flow run.

Reply:
ah, that makes sense. thanks